packages/block-library/src/comment-template/edit.js

```diff
@@ -23,11 +28,92 @@ const TEMPLATE = [
 	[ 'core/comment-edit-link' ],
 ];
 
-function CommentTemplateInnerBlocks() {
-	const innerBlocksProps = useInnerBlocksProps( {}, { template: TEMPLATE } );
-	return <li { ...innerBlocksProps } />;
+/**
+ * Component which renders the inner blocks of the Comment Template.
+ *
+ * @param {Object} props                         Component props.
+ * @param {Array}  [props.blockContext]          - A comment object.
+ * @param {Array}  [props.activeBlockContext]    - The block that is currently active.
+ * @param {Array}  [props.setActiveBlockContext] - The setter for activeBlockContext.
+ * @param {Array}  [props.firstBlock]            - First comment in the array.
+ * @param {Array}  [props.blocks]                - Array of blocks returned from
+ *                                               getBlocks() in parent .
+ * @return {WPElement}                 		Inner blocks of the Comment Template
+ */
+function CommentTemplateInnerBlocks( {
+	blockContext,
+	activeBlockContext,
+	setActiveBlockContext,
+	firstBlock,
+	blocks,
+} ) {
+	const { children, ...innerBlocksProps } = useInnerBlocksProps(
+		{},
+		{ template: TEMPLATE }
+	);
+	return (
+		<li { ...innerBlocksProps }>
+			{ blockContext === ( activeBlockContext || firstBlock ) ? (
+				children
+			) : (
+				<BlockPreview
+					blocks={ blocks }
+					__experimentalLive
+					__experimentalOnClick={ () =>
+						setActiveBlockContext( blockContext )
+					}
+				/>
+			) }
+			{ blockContext?.children?.length > 0 ? (
+				<CommentsList
+					blockContexts={ blockContext.children }
+					activeBlockContext={ activeBlockContext }
+					setActiveBlockContext={ setActiveBlockContext }
+					blocks={ blocks }
+				/>
+			) : null }
+		</li>
+	);
 }
 
+/**
+ * Component that renders a list of (nested) comments. It is called recursively.
+ *
+ * @param {Object} props                         Component props.
+ * @param {Array}  [props.blockProps]            - Props from parent's `useBlockProps()`.
+ * @param {Array}  [props.blockContexts]         - Array of comment objects.
```

Comment:
`blockContexts` is a bit confusing here. It's the tree structure representing the structure of comments with the block contexts for each of them, right?

Reply:
That's right. Yeah, I've re-used the naming from the Query Loop block here to be honest, but I agree that it's not super clear, even I got confused once or twice. 

I think it would be better to rename it to `comments` or `commentsList` or `commentsTree` and the _individual_ items in that list/tree could can be simply `comment`. What do you think?

Reply:
Alright, Renamed `blockContexts` and related blocks to simply `comments`. I think it's clearer this way 🙂